packages/rspeedy/core/src/plugins/target.plugin.ts

```diff
@@ -5,13 +5,22 @@
 import type { RsbuildPlugin } from '@rsbuild/core'
 
 import { getESVersionTarget } from '../utils/getESVersionTarget.js'
+import { isWeb } from '../utils/is-web.js'
 
 export function pluginTarget(): RsbuildPlugin {
   return {
     name: 'lynx:rsbuild:target',
     setup(api) {
-      api.modifyBundlerChain((options) => {
-        options.target([getESVersionTarget()])
+      api.modifyBundlerChain((options, { environment }) => {
+        if (isWeb(environment)) {
+          options.target([
+            getESVersionTarget(),
+            // Add `target: 'web'` to make Rsbuild inject HMR related code.
+            'web',
```

Comment:
We will keep using `'web'` instead of `'webworker'` since Rsbuild would currently only inject HMR entry on `'web'`.

See: https://github.com/web-infra-dev/rsbuild/blob/09e2b8eaa80965c3ff6c93fe72a0942e5b473e69/packages/core/src/server/compilationMiddleware.ts#L16